gam/gam.py

```diff
@@ -44,14 +47,41 @@ def __init__(
         self,
         k=2,
         attributions_path="local_attributions.csv",
+        attributions=None,
+        feature_labels=None,
         cluster_method=None,
         distance="spearman",
         use_normalized=True,
         scoring_method=None,
         max_iter=100,
         tol=1e-3,
     ):
+
         self.attributions_path = attributions_path
+
+        # preprocessing attributions and feature_labels if they're a pandas dataframe, numpy array, or list
+        if isinstance(attributions, pd.DataFrame):
```

Comment:
Can you add some tests around this section - I think it would be helpful to test:

- the new functionality of in-memory attributions (on a dummy pandas data frame - probably created from [test_attributes.csv](https://github.com/capitalone/global-attribution-mapping/blob/master/tests/test_attributes.csv) in the test folder)

- the conditionals around the different types of inputs, specifically if you could write a test that purposefully makes this section throw this exception:  
```raise ValueError("You must have both 'attributions' and 'feature_labels' if 'attributions' is not a dataframe.")```
 You can use [pytest.assertRaises](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions) for this

It also may be worth breaking this large conditional into a helper function just to make it more readable/testsable